rotkehlchen/exchanges/manager.py

```diff
@@ -292,14 +292,11 @@ def initialize_exchanges(
                 )
                 self.connected_exchanges[location].append(exchange_obj)
 
-    def get_all_binance_pairs(self) -> List[str]:
-        if Location.BINANCE in self.connected_exchanges:
-            binance = cast('Binance', self.connected_exchanges[Location.BINANCE][0])
-            return list(binance.symbols_to_pair.keys())
-        if Location.BINANCEUS in self.connected_exchanges:
-            binance = cast('Binance', self.connected_exchanges[Location.BINANCEUS][0])
-            return list(binance.symbols_to_pair.keys())
-        return []
+    def get_all_binance_pairs(self, location: Location) -> List[str]:
+        pairs = list(query_binance_exchange_pairs(location=location).keys())
```

Comment:
I suggest you do it differently.  Just like we do with all other api endpoints the error should be handled by the api. So write down what exceptions can be raised or return None if you don't want to raise exceptions. And then instead of using the self.msg_aggregator here, just return an api error.